cns/restserver/internalapi_test.go

```diff
@@ -48,6 +48,60 @@ func TestCreateOrUpdateNCWithLargerVersionComparedToNMAgent(t *testing.T) {
 	validateCreateOrUpdateNCInternal(t, 2, "1")
 }
 
+func TestCreateAndUpdateNCWithSecondaryIPNCVersion(t *testing.T) {
+	restartService()
+
+	setEnv(t)
+	setOrchestratorTypeInternal(cns.KubernetesCRD)
+	// NC version set as 0 which is the default initial value.
+	ncVersion := 0
+	secondaryIPConfigs := make(map[string]cns.SecondaryIPConfig)
+	receivedSecondaryIPConfigs := make(map[string]cns.SecondaryIPConfig)
+	ncID := "testNc1"
+
+	// Build secondaryIPConfig, it will have one item as {IPAddress:"10.0.0.16", NCVersion: 0}
+	ipAddress := "10.0.0.16"
+	secIPConfig := newSecondaryIPConfig(ipAddress, ncVersion)
+	ipId := uuid.New()
+	secondaryIPConfigs[ipId.String()] = secIPConfig
+	req := createNCReqInternal(t, secondaryIPConfigs, ncID, strconv.Itoa(ncVersion))
+	containerStatus := svc.state.ContainerStatus[req.NetworkContainerid]
+	// Validate secondary IPs' NC version has been updated by NC request
+	receivedSecondaryIPConfigs = containerStatus.CreateNetworkContainerRequest.SecondaryIPConfigs
+	for _, secIPConfig := range receivedSecondaryIPConfigs {
+		// Though "10.0.0.16" IP exists in NC version 1, secodanry IP still keep its original NC version 0
+		if secIPConfig.IPAddress != "10.0.0.16" || secIPConfig.NCVersion != 0 {
+			t.Fatalf("nc request version is %d, secondary ip %s nc version is %d, expected nc version is 0",
+				ncVersion, secIPConfig.IPAddress, secIPConfig.NCVersion)
+		}
+	}
+
+	// now Validate Update, simulate the CRD status where have 2 IP addresses as "10.0.0.16" and "10.0.0.17" with NC version 1
+	// The secondaryIPConfigs build from CRD will be {[IPAddress:"10.0.0.16", NCVersion: 1,]; [IPAddress:"10.0.0.17", NCVersion: 1,]}
+	// However, when CNS saveNetworkContainerGoalState, since "10.0.0.16" already with NC version 0, it will set it back to 0.
+	ncVersion++
+	secIPConfig = newSecondaryIPConfig(ipAddress, ncVersion)
+	// "10.0.0.16" will be update to NC version 1 in CRD, reuse the same uuid with it.
+	secondaryIPConfigs[ipId.String()] = secIPConfig
+
+	// Add {IPAddress:"10.0.0.17", NCVersion: 1} in secondaryIPConfig
+	ipAddress = "10.0.0.17"
+	secIPConfig = newSecondaryIPConfig(ipAddress, ncVersion)
+	ipId = uuid.New()
+	secondaryIPConfigs[ipId.String()] = secIPConfig
+	req = createNCReqInternal(t, secondaryIPConfigs, ncID, strconv.Itoa(ncVersion))
+	// Validate secondary IPs' NC version has been updated by NC request
+	receivedSecondaryIPConfigs = containerStatus.CreateNetworkContainerRequest.SecondaryIPConfigs
+	for _, secIPConfig := range receivedSecondaryIPConfigs {
+		// Though "10.0.0.16" IP exists in NC version 1, secodanry IP still keep its original NC version 0
+		if (secIPConfig.IPAddress == "10.0.0.16" && secIPConfig.NCVersion != 0) ||
+			(secIPConfig.IPAddress == "10.0.0.17" && secIPConfig.NCVersion != 1) {
+			t.Fatalf("nc request version is %d, secondary ip %s nc version is %d, expected nc version is 0",
```

Comment:
prefer a switch statement here, and messages more specific to each. in case the inconsistent version was with 10.0.0.17, the message still tells you that it expected 0, which is not appropriate:
```go
switch secIPConfig:
case "10.0.0.16":
    // check for 0
case "10.0.0.17":
    // check for 1
default:
    // error since this is unexpected

```